serveradmin/apps/models.py

```diff
@@ -45,3 +51,66 @@ def set_disabled(sender, instance, **kwargs):
     """
     if not instance.is_active:
         Application.objects.filter(owner=instance).update(disabled=True)
+
+
+class PublicKey(models.Model):
+    application = models.ForeignKey(
+        Application, related_name="public_keys", on_delete=models.CASCADE
+    )
+    key_algorithm = models.CharField(max_length=80)
+    key_base64 = models.CharField(primary_key=True, max_length=1024)
+    key_comment = models.CharField(max_length=80)
+
+    def __str__(self):
+        if not (self.key_algorithm and self.key_base64):
+            return self.key_comment
+        # This format is used by ssh-add -l
+        blob = self.load().asbytes()
+        return 'SHA256:' + b64encode(sha256(blob).digest()).decode()
```

Comment:
This is only for displaying the applications list overview in the admin ui, what's there to feel uncomfortable about? There is a digest function, but it's not what is displayed by ssh-agent. I find this more helpful as this makes it easy to correlate keys in your agent with apps in serveradmin.

Reply:
ok true, you get a sha256 base64 string per default (at least on our machines). i am used to md5 though (add `-Emd5` to `ssh-add -l`). that is also what is displayed on github for example. i just don't like to write any such code myself because there is most probably never a real advantage for it. but go for it. this is not a blocker

Reply:
I'll keep it, maybe we can even upstream this to paramiko.